source/index.html.md

```diff
@@ -699,6 +700,23 @@ object][ApplePay-PaymentToken] for more information.
   </dt>
   <dd>
     Expiry year of card to charge.
+  </dd>
+    <dt>mobilepayonline[cryptogram]
+    <span class="type">[:base64:]{19,21}</span>
+  </dt>
+  <dd>
+    Cryptogram extracted from from MobilePay Online token data. Required on
+    token-based authentication.
+    <div class="type">Optional</div>
+  </dd>
+  </dd>
+    <dt>mobilepayonline[eci]
+    <span class="type">[0-9]{2}</span>
+  </dt>
+  <dd>
+    Electronic Commerce Indicator (ECI) extracted from the MobilePay Online token.
+    data. Only allowed on token-based authentication.
+    <div class="type">Optional</div>
```

Comment:
I suspect it's required for Visa?

Reply:
This is the classic very sparse API docs phenomenon at work. If we where to also receive Mastercard in the minimalistic way, we would only receive a `cryptogram` and not `eci`. And instead of getting bugged down in _if this then that, unless a, b, c but only if also d_ I have left it optional and the gateway will be the one responding with _"ECI missing on Visa, bla bla bla"_. 

This is checked by `check_threed_secure_data_for_visa()`
https://github.com/clearhaus/klovedal/blob/master/lib/klovedal/filter/payment_method/token_common.rb#L47-L49